producer/kinesis.go

```diff
@@ -123,6 +134,8 @@ func (prod *Kinesis) Configure(conf core.PluginConfig) error {
 
 	prod.streamMap = conf.GetStreamMap("StreamMapping", "default")
 	prod.batch = core.NewMessageBatch(conf.GetInt("BatchMaxMessages", 500))
+	prod.recordMaxMessages = conf.GetInt("RecordMaxMessages", 1)
```

Comment:
Give an error if RecordMaxMessages is == 0 or force set it to shared.MaxI(1,conf.GetInt("RecordMaxMessages", 1)).
Otherwise there could be a divide by 0 later.